src/MongoDB.Driver.Core/Core/Connections/TcpStreamFactory.cs

```diff
@@ -179,7 +179,7 @@ private async Task ConnectAsync(Socket socket, EndPoint endPoint, CancellationTo
                 {
                     var dnsEndPoint = endPoint as DnsEndPoint;
 #if NETSTANDARD1_5 || NETSTANDARD1_6
-                    await Task.Run(() => socket.Connect(endPoint)); // TODO: honor cancellationToken
+                    await Task.Run(() => socket.Connect(endPoint)).ConfigureAwait(false); // TODO: honor cancellationToken
```

Comment:
This seems strange to be included in this commit.  What's this about?

Reply:
I believe Robert made this this change so that Driver.Core would build with zero warnings. The project treats warnings as errors, and I think we've required all usages of `await` need a matching`ConfigureAwait(...)`

Reply:
Vincent is correct.

I was trying to not make any code changes but there was just no way we could relax this rule, hence the code fix instead.